Textfield/Textfield.jsx

```diff
@@ -64,33 +68,58 @@ class TextfieldInput extends MaterialComponent {
       }
     );
   }
+
   componentWillUnmount() {
     this.MDComponent && this.MDComponent.destroy && this.MDComponent.destroy();
   }
-  materialDom(allprops) {
+  
+  _onFocus() {
+    this.setState({
+      isFocused: true
+    });
+  }
+
+  _onBlur() {
+    this.setState({
+      isFocused: false
+    });
+  }
+
+  materialDom(allprops, { showFloatingLabel, isFocused }) {
     let { className, ...props } = allprops;
+    let labelClassName = "mdc-textfield__label";
+
+    const upgrade = props.value || isFocused;
```

Comment:
so upgrade is not the function of value or focus, its just if the js enhancement has been done or not